pkg/controller/master/setting/rancher_manager_support.go

```diff
@@ -0,0 +1,83 @@
+package setting
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+
+	ranchersettings "github.com/rancher/rancher/pkg/settings"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+)
+
+const (
+	rancherName   = "Rancher"
+	harvesterName = "Harvester"
+)
+
+// syncRancherManagerSupport updates ui-pl and ui-brand setting in rancher
+// to match the value of rancher-manager-support setting
+// if rancher-manager-support is true, ui-pl value is set to "Rancher" and ui-brand default is set to ""
+// if rancher-manager-support is false, ui-pl value is set to "Harvester" and ui-brand default is set to "harvester"
+func (h *Handler) syncRancherManagerSupport(setting *harvesterv1.Setting) error {
+	value := setting.Value
+	if value == "" {
+		value = setting.Default
+	}
+
+	enable, err := strconv.ParseBool(value)
+	if err != nil {
+		return err
+	}
+
+	err = h.updateUIPL(enable)
+	if err != nil {
+		return err
+	}
+	err = h.updateUIBrand(enable)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func (h *Handler) updateUIPL(enable bool) error {
+	uiPL, err := h.rancherSettingCache.Get(ranchersettings.UIPL.Name)
+	if err != nil {
+		return err
+	}
+
+	uiPLCopy := uiPL.DeepCopy()
+	if enable {
```

Comment:
the default value of the setting is already `rancher` and is applied with left empty.

should we following the same logic as we do for ui-brand and leave it empty when enabled?

On an existing install they are already empty.

```
(⎈|local:N/A)➜  harvester git:(update-ui-pl) ✗ k get settings.management ui-pl ui-brand
NAME       VALUE
ui-pl
ui-brand
```

Reply:
the default value of ui-pl in Harvester is patched to "Harvester" in pkg/controller/master/rancher/embedded_rancher.go

Reply:
The UI currently uses this ui-pl default value "Harvester" to decide whether to display the bootstrap password  box on the first login